# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Recover.h
# Commit: 375186706cae
# Full Hash: 375186706caeb4d480bb8c7be78a349601a3e7c1
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-02-13 09:52:34
# Regressor Bug: 1688033
# File Overlap Count: 1
# Description:
#   Bug 1688033: Recover CreateArgumentsObject on bailout r=nbp
#   
#   The original version of this patch only recovered CreateArgumentsObject on bailout if it didn't escape, but I removed that code based on discussion in #warpbuilder.
#   
#   This code is tested by `/ion/dce-with-rinstructions.js`. Prior to this patch, it failed with `--scalar-replace-arguments`.
# ==============================================================================

diff -r adaf366a60fa -r 375186706cae js/src/jit/Recover.h
--- a/js/src/jit/Recover.h	Fri Feb 12 20:28:47 2021 +0000
+++ b/js/src/jit/Recover.h	Fri Feb 12 20:28:48 2021 +0000
@@ -132,6 +132,7 @@
   _(AtomicIsLockFree)          \
   _(BigIntAsIntN)              \
   _(BigIntAsUintN)             \
+  _(CreateArgumentsObject)     \
   _(AssertRecoveredOnBailout)
 
 class RResumePoint;
@@ -870,6 +871,14 @@
                              SnapshotIterator& iter) const override;
 };
 
+class RCreateArgumentsObject final : public RInstruction {
+ public:
+  RINSTRUCTION_HEADER_NUM_OP_(CreateArgumentsObject, 1)
+
+  [[nodiscard]] bool recover(JSContext* cx,
+                             SnapshotIterator& iter) const override;
+};
+
 class RAssertRecoveredOnBailout final : public RInstruction {
  public:
   RINSTRUCTION_HEADER_NUM_OP_(AssertRecoveredOnBailout, 1)